.github/workflows/tests.yml

```diff
@@ -61,9 +63,66 @@ jobs:
         git config --global url."https://github.com/".insteadOf ssh://git@github.com/
         npm run install:ci
 
+  build:
+    name: Build application & cache it
+    runs-on: ubuntu-latest
+    needs: setup
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with:
+          path: ~/.asdf
+          key: ${{ runner.os }}-asdf-v3-${{ hashFiles('.tool-versions') }}
+      - uses: mbta/actions/reshim-asdf@v1
+      - uses: actions/cache@v2
+        with:
+          path: deps
+          key: ${{ runner.os }}-mix-${{ hashFiles('**/mix.lock') }}
+      - uses: actions/cache@v2
+        with:
+          path: "apps/site/*/node_modules"
+          key: ${{ runner.os }}-nodejs-${{ hashFiles('apps/site/*/package-lock.json') }}
+      - name: Cache react_renderer output
+        uses: actions/cache@v2
+        with:
+          path: apps/site/react_renderer/dist/app.js
+          key: ${{ runner.os }}-react_renderer-${{ hashFiles('apps/site/react_renderer/package-lock.json') }}
+        id: react_renderer-cache
+      - run: npm run --prefix apps/site/react_renderer build
+        if: steps.react_renderer-cache.outputs.cache-hit != 'true'
+      
+      - name: Cache front end assets
+        uses: actions/cache@v2
+        with:
+          path: apps/site/priv/static/js
+          key: ${{ runner.os }}-assets-${{ hashFiles('apps/site/assets/package-lock.json') }}
+        id: assets-cache
+      - run: npm run --prefix apps/site/assets webpack:build
+        if: steps.assets-cache.outputs.cache-hit != 'true'
+
+      - name: Cache back end assets
+        uses: actions/cache@v2
+        with:
+          path: apps/site/priv/static
+          key: ${{ runner.os }}-digest-${{ hashFiles('apps/site/assets/package-lock.json') }}
+        id: digest-cache
+      - run: mix phx.digest
+        if: steps.digest-cache.outputs.cache-hit != 'true'
+      
+      - name: Cache _build
+        uses: actions/cache@v2
+        with:
+          path: _build
+          key: ${{ runner.os }}-build-${{ hashFiles('**/mix.lock') }}-${{ hashFiles('apps/site/*/package-lock.json') }}
```

Comment:
I think you mentioned this in Slack or Asana, but this also needs to hash the files themselves, since they're getting compiled as a part of the next step.

Reply:
d'oh, that's completely right and it slipped my mind! thank you

Reply:
the workflow's been reworked a bit, but I'm hoping `${{ runner.os }}-build-cache-${{ hashFiles('**/lib/**.ex', 'apps/site/priv/static/**', '**/mix.lock') }}` might suffice as a useful cache key for the `_build` folder?